src/platform/Linux/MdnsImpl.cpp

```diff
@@ -579,13 +579,16 @@ void MdnsAvahi::HandleBrowse(AvahiServiceBrowser * browser, AvahiIfIndex interfa
         if (strcmp("local", domain) == 0)
         {
             MdnsService service = {};
+            char typeAndProtocol[kMdnsTypeAndProtocolMaxSize + 1];
 
             strncpy(service.mName, name, sizeof(service.mName));
-            strncpy(service.mType, type, sizeof(service.mType));
             service.mName[kMdnsNameMaxSize] = 0;
+            strncpy(typeAndProtocol, type, sizeof(typeAndProtocol));
```

Comment:
why do we need a separate copy?

The original code is suspect because it uses a local variable probably due to const-correctness. However getting the protocol from type seems totally doable in `const char *`.